python/update_package_properties.py

```diff
@@ -0,0 +1,346 @@
+
+
+def get_debian_trailer(microsoft_email: str, name_surname: str, changelog_date: date):
+    formatted_date = changelog_date.strftime("%a, %d %b %Y %H:%M:%S %z ")
+    return f" -- {name_surname} <{microsoft_email}>  {formatted_date} \n "
+
+
+def convert_citus_changelog_into_rpm_changelog(changelog_params: ChangelogParams) -> str:
+    header = get_rpm_header(changelog_params)
+    rpm_changelog = f"{header.strip()}\n- Official {changelog_params.get_project_version()} release of {changelog_params.get_project_name().capitalize()}"
+
+    return rpm_changelog
+
+
+def update_rpm_spec(changelog_param: ChangelogParams, spec_file_name: str,
+                    templates_path: str) -> None:
+    env = get_template_environment(templates_path)
+
+    rpm_version = get_rpm_version(changelog_param.get_project_name(), changelog_param.get_project_version())
+    template = env.get_template('project.spec.tmpl')
+    rpm_changelog_history = get_rpm_changelog_history(spec_file_name)
+
+    history_lines = rpm_changelog_history.splitlines()
+
+    if len(history_lines) > 0 and changelog_param.get_project_version() in history_lines[1]:
+        raise ValueError(f"{changelog_param.get_project_version()} already exists in rpm spec file")
+
+    latest_changelog = convert_citus_changelog_into_rpm_changelog(changelog_param)
+    changelog = f"{latest_changelog}\n\n{rpm_changelog_history}"
+    content = template.render(version=changelog_param.get_project_version(), rpm_version=rpm_version,
+                              project_name=changelog_param.get_project_name(),
+                              fancy_version_no=changelog_param.get_fancy_version_number(), changelog=changelog)
+    with open(spec_file_name, "w+") as writer:
+        writer.write(content)
+
+
+@validate_parameters
+def update_all_changes(github_token: non_empty(non_blank(str)), project_name: non_empty(str),
+                       project_version: is_version(str),
+                       tag_name: is_tag(str), fancy: bool, fancy_version_number: non_negative(int),
+                       microsoft_email: is_email(str),
+                       name_surname: non_empty(non_blank(str)), release_date: datetime, packaging_path: str):
+    templates_path = f"{BASE_PATH}/templates"
+    update_pkgvars(project_name, project_version, fancy, fancy_version_number, templates_path, f"{packaging_path}")
+    latest_changelog = get_changelog_for_tag(github_token, project_name, tag_name)
+    changelog_param = ChangelogParams()
+    changelog_param.set_project_version(project_version).set_project_name(project_name).set_microsoft_email(
+        microsoft_email).set_name_surname(name_surname).set_fancy_version_number(
+        fancy_version_number).set_changelog_date(release_date).set_fancy(fancy).set_latest_changelog(latest_changelog)
```

Comment:
Having these getters and setters seems a bit unnecesary, since you are only setting these fields once you can use the `__init__` method of the class:
```suggestion
    changelog_param = ChangelogParams(
         project_version=project_version,
         project_name=project_name,
         microsoft_email=microsoft_email,
         name_surname=name_surname,
         fancy_version_number=fancy_version_number,
         release_date=release_date,
         fancy=fancy,
         latest_changelog=latest_changelog,
    )
```

Reply:
Fixed
https://github.com/citusdata/tools/commit/be5591732f65aef151977d4523eb5ba560f5d5fd